ambersim/utils/io_utils.py

```diff
@@ -0,0 +1,237 @@
+from pathlib import Path
+from typing import List, Optional, Tuple, Union
+
+import coacd
+import mujoco as mj
+import numpy as np
+import trimesh
+from dm_control import mjcf
+from mujoco import mjx
+
+from ambersim import ROOT
+
+
+def _check_filepath(filepath: Union[str, Path]) -> str:
+    """Checks validity of a filepath for model loading."""
+    assert isinstance(filepath, (str, Path))
+
+    # checking whether file exists
+    if isinstance(filepath, str):
+        filepath = Path(filepath)  # global/local
+    if not filepath.exists():
+        filepath = ROOT / filepath  # repo root
+        if not filepath.exists():
+            raise ValueError("The model file doesn't exist at the specified path!")
+    filepath = str(filepath)
+    return filepath
+
+
+def _rmtree(f: Path):
+    """Recursively deletes a directory using pathlib.
+
+    See: https://stackoverflow.com/a/66552066
+    """
+    if f.is_file():
+        f.unlink()
+    else:
+        for child in f.iterdir():
+            _rmtree(child)
+        f.rmdir()
+
+
+# ############# #
+# MODEL LOADING #
+# ############# #
+
+
+def _modify_robot_float_base(filepath: Union[str, Path]) -> mj.MjModel:
+    """Modifies a robot to have a floating base if it doesn't already."""
+    # loading current robot
+    assert str(filepath).split(".")[-1] == "xml"
+    robot = mjcf.from_file(filepath, model_dir="/".join(filepath.split("/")[:-1]))
+
+    # only add free joint if the first body after worldbody has no freejoints
+    assert robot.worldbody is not None
+    if robot.worldbody.body[0].freejoint is None:
+        # creates a new robot object and attaches the current robot to it freely
+        arena = mjcf.RootElement(model=robot.model)
+        attachment_frame = arena.attach(robot)
+        attachment_frame.add("freejoint", name="freejoint")
+        robot = arena
+
+        # ensuring that the base link has inertial attributes
```

Comment:
@kli58 (I moved this comment here for readability)

Minor comment on ```mj_to_mjx_model```: this read as only returning ```mjx_mode```l instead of (```mjx_model,mjx_data```) pair